pkg/tfgen/docs_test.go

```diff
@@ -846,6 +853,51 @@ func TestFindRepoPath(t *testing.T) {
 	assert.Equal(t, "./terraform-provider-bar", actual)
 }
 
+func TestExampleGeneration(t *testing.T) {
+	info := testprovider.ProviderMiniRandom()
+
+	markdown := []byte(`
+## Examples
+
+There is some more code in here.
+
+~~~java
+throw new Exception("!");
+~~~
+`)
+
+	markdown = bytes.ReplaceAll([]byte(markdown), []byte("~~~"), []byte("```"))
+
+	info.Resources["random_integer"].Docs = &tfbridge.DocInfo{
+		Markdown: markdown,
+	}
+
+	inmem := afero.NewMemMapFs()
+
+	g, err := NewGenerator(GeneratorOptions{
+		Package:      info.Name,
+		Version:      info.Version,
+		Language:     Schema,
+		ProviderInfo: info,
+		Root:         inmem,
+		Sink: diag.DefaultSink(io.Discard, io.Discard, diag.FormatOptions{
+			Color: colors.Never,
+		}),
+	})
+	assert.NoError(t, err)
+
+	err = g.Generate()
+	assert.NoError(t, err)
+
+	f, err := inmem.Open("schema.json")
+	assert.NoError(t, err)
+
+	schemaBytes, err := io.ReadAll(f)
+	assert.NoError(t, err)
+
+	assert.NotContains(t, string(schemaBytes), "{{% //examples %}}")
```

Comment:
Deep in the schema.json file we now have the rendered doc for the random_integer resource to assert against.